client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx

```diff
@@ -125,6 +130,15 @@ const SiteSetupList = ( {
 		}
 	}, [ currentTaskId, dispatch, tasks ] );
 
+	// If specified, then automatically complete the current task when viewed
+	// if it is not already complete.
+	useEffect( () => {
+		if ( currentTask?.completeOnView && ! currentTask.isCompleted ) {
```

Comment:
minor: should we check for `isCompleted` in the same way that `completeOnView`? Feel free to ignore it.

Reply:
Actually the api response contains `isCompleted` so I think it would be an overkill (whereas `completeOnView` is added later and may be `undefined`)

Reply:
Harris is right but I've just done it anyway.